From 9d28e2e87840c4373b4bfa2583135a6c729e4d22 Mon Sep 17 00:00:00 2001
From: Paras Sharma <paras.sharma@oss.qualcomm.com>
Date: Tue, 2 Dec 2025 12:18:47 +0530
Subject: [PATCH 07/15] ANDROID: qcom: Add PCIe framework related symbols

Add PCIe ECAM and PCIe core framework provided exported APIs related
symbols for PCIe passthrough related driver usage.

Symbols added:
pci_bus_add_device
pci_bus_add_devices
pci_bus_assign_resources
pci_bus_claim_resources
pci_bus_size_bridges
pci_ecam_free
pci_flags
pci_scan_root_bus_bridge
pci_stop_and_remove_bus_device_locked
pcie_bus_configure_settings

Bug: 465316788
Bug: 469634950
Change-Id: I208b6b6023609e440efb99640c8e27efc103147e
Signed-off-by: Paras Sharma <paras.sharma@oss.qualcomm.com>
(cherry picked from commit 0289b94af2820b1733887372226c6208343254a0)
---
 gki/aarch64/abi.stg      | 100 +++++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/qcom |  10 ++++
 2 files changed, 110 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 1b4487b5a13c..da20c51a5207 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -373311,6 +373311,11 @@ function {
   return_type_id: 0x48b5725f
   parameter_id: 0x0206e690
 }
+function {
+  id: 0x101bdac8
+  return_type_id: 0x48b5725f
+  parameter_id: 0x020e2d6c
+}
 function {
   id: 0x101c2d25
   return_type_id: 0x48b5725f
@@ -384305,6 +384310,11 @@ function {
   parameter_id: 0x347303b4
   parameter_id: 0x3e10b518
 }
+function {
+  id: 0x1d64d869
+  return_type_id: 0x48b5725f
+  parameter_id: 0x37f227e9
+}
 function {
   id: 0x1d657c6d
   return_type_id: 0x48b5725f
@@ -487805,6 +487815,42 @@ elf_symbol {
   type_id: 0xfaddfa97
   full_name: "pci_ats_supported"
 }
+elf_symbol {
+  id: 0xad049f3d
+  name: "pci_bus_add_device"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x466c6b90
+  type_id: 0x14e1f000
+  full_name: "pci_bus_add_device"
+}
+elf_symbol {
+  id: 0x082c11eb
+  name: "pci_bus_add_devices"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x08860e11
+  type_id: 0x1d64d869
+  full_name: "pci_bus_add_devices"
+}
+elf_symbol {
+  id: 0x375c1ee2
+  name: "pci_bus_assign_resources"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x600ecd92
+  type_id: 0x1d64d869
+  full_name: "pci_bus_assign_resources"
+}
+elf_symbol {
+  id: 0x2ad69fc2
+  name: "pci_bus_claim_resources"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x78ecf14e
+  type_id: 0x185a3adc
+  full_name: "pci_bus_claim_resources"
+}
 elf_symbol {
   id: 0xbec826c0
   name: "pci_bus_read_config_dword"
@@ -487823,6 +487869,15 @@ elf_symbol {
   type_id: 0xcd829c8f
   full_name: "pci_bus_resource_n"
 }
+elf_symbol {
+  id: 0x22e85b74
+  name: "pci_bus_size_bridges"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x98beef86
+  type_id: 0x185a3adc
+  full_name: "pci_bus_size_bridges"
+}
 elf_symbol {
   id: 0x1540c319
   name: "pci_bus_type"
@@ -487967,6 +488022,15 @@ elf_symbol {
   type_id: 0x9ec927c4
   full_name: "pci_ecam_create"
 }
+elf_symbol {
+  id: 0x0df090d1
+  name: "pci_ecam_free"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x58fb63c3
+  type_id: 0x101bdac8
+  full_name: "pci_ecam_free"
+}
 elf_symbol {
   id: 0x0872d05c
   name: "pci_ecam_map_bus"
@@ -488156,6 +488220,15 @@ elf_symbol {
   type_id: 0x578fa618
   full_name: "pci_find_next_capability"
 }
+elf_symbol {
+  id: 0xbbf8a63b
+  name: "pci_flags"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0xf296206e
+  type_id: 0x4585663f
+  full_name: "pci_flags"
+}
 elf_symbol {
   id: 0x08190210
   name: "pci_free_irq"
@@ -488624,6 +488697,15 @@ elf_symbol {
   type_id: 0x99f942bc
   full_name: "pci_save_state"
 }
+elf_symbol {
+  id: 0x18af963b
+  name: "pci_scan_root_bus_bridge"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x604ec397
+  type_id: 0x93acae9b
+  full_name: "pci_scan_root_bus_bridge"
+}
 elf_symbol {
   id: 0xc62d75dd
   name: "pci_select_bars"
@@ -488813,6 +488895,15 @@ elf_symbol {
   type_id: 0xfaddfa97
   full_name: "pcie_aspm_enabled"
 }
+elf_symbol {
+  id: 0x4f554530
+  name: "pcie_bus_configure_settings"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x300ea93b
+  type_id: 0x185a3adc
+  full_name: "pcie_bus_configure_settings"
+}
 elf_symbol {
   id: 0xb0cf8b54
   name: "pcie_capability_clear_and_set_word_locked"
@@ -523826,8 +523917,13 @@ interface {
   symbol_id: 0xbab6c772
   symbol_id: 0x66f0c8cd
   symbol_id: 0xb10958e7
+  symbol_id: 0xad049f3d
+  symbol_id: 0x082c11eb
+  symbol_id: 0x375c1ee2
+  symbol_id: 0x2ad69fc2
   symbol_id: 0xbec826c0
   symbol_id: 0x45091f48
+  symbol_id: 0x22e85b74
   symbol_id: 0x1540c319
   symbol_id: 0x06548b7e
   symbol_id: 0xe99b6d83
@@ -523844,6 +523940,7 @@ interface {
   symbol_id: 0x01a00013
   symbol_id: 0xb9d0b3a8
   symbol_id: 0x2b8efaa9
+  symbol_id: 0x0df090d1
   symbol_id: 0x0872d05c
   symbol_id: 0x96b46225
   symbol_id: 0x35a3d772
@@ -523865,6 +523962,7 @@ interface {
   symbol_id: 0x780d1110
   symbol_id: 0xbbb674cc
   symbol_id: 0xdea420f5
+  symbol_id: 0xbbf8a63b
   symbol_id: 0x08190210
   symbol_id: 0x8ffabaa9
   symbol_id: 0x50bce06e
@@ -523917,6 +524015,7 @@ interface {
   symbol_id: 0xd2a1df18
   symbol_id: 0x18bfee32
   symbol_id: 0x53d0da35
+  symbol_id: 0x18af963b
   symbol_id: 0xc62d75dd
   symbol_id: 0x9595d229
   symbol_id: 0xfc86cde9
@@ -523938,6 +524037,7 @@ interface {
   symbol_id: 0x52200b18
   symbol_id: 0xfda7b705
   symbol_id: 0xefd6d09b
+  symbol_id: 0x4f554530
   symbol_id: 0xb0cf8b54
   symbol_id: 0x5092d9d9
   symbol_id: 0xf1513a2c
diff --git a/gki/aarch64/symbols/qcom b/gki/aarch64/symbols/qcom
index 0c89c770dd2f..9b45fb5f02f7 100644
--- a/gki/aarch64/symbols/qcom
+++ b/gki/aarch64/symbols/qcom
@@ -2245,6 +2245,11 @@
   pci_aer_clear_nonfatal_status
   pci_alloc_irq_vectors_affinity
   pci_assign_resource
+  pci_bus_add_device
+  pci_bus_add_devices
+  pci_bus_assign_resources
+  pci_bus_claim_resources
+  pci_bus_size_bridges
   pci_bus_type
   pci_clear_master
   pci_dev_get
@@ -2255,11 +2260,13 @@
   pci_disable_device
   pci_disable_msi
   pci_ecam_create
+  pci_ecam_free
   pci_ecam_map_bus
   pcie_capability_clear_and_set_word_unlocked
   pcie_capability_read_word
   pci_enable_device
   pci_find_ext_capability
+  pci_flags
   pci_free_irq_vectors
   pci_generic_config_read
   pci_generic_config_write
@@ -2282,13 +2289,16 @@
   pci_request_region
   pci_restore_state
   pci_save_state
+  pci_scan_root_bus_bridge
   pci_set_master
   pci_set_power_state
+  pci_stop_and_remove_bus_device_locked
   pci_store_saved_state
   pci_unregister_driver
   pci_walk_bus
   pci_write_config_dword
   pci_write_config_word
+  pcie_bus_configure_settings
   pcpu_alloc_noprof
   pcpu_nr_pages
   __percpu_down_read
-- 
2.50.1

